src/services/docling_service.py

```
@@ -0,0 +1,203 @@
+            logger.error("Docling result retrieval failed", task_id=task_id, error=str(e))
+            raise
+
+    async def _poll_result(self, client: httpx.AsyncClient, task_id: str, poll_interval: float, timeout: float) -> Dict[str, Any]:
+        """Internal polling logic."""
+        elapsed = 0.0
+        while elapsed < timeout:
+            try:
+                response = await client.get(f"{self.docling_url}/v1/status/poll/{task_id}")
+                response.raise_for_status()
+                status_data = response.json()
+            except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)